File: 0152. Maximum Product Subarray/Solution.java
// brute force Solution 

// class Solution {
//     public int maxProduct(int[] nums) {
//         int prod = nums[0];

//         for(int i=0;i<nums.length;i++){
//             int subProd = 1;
//             for(int j=i;j<nums.length;j++){
//                 subProd = subProd*nums[j];
//                 prod = Math.max(subProd,prod);
//             }   
//         }

//         return prod;
//     }
// }

File: 0152. Maximum Product Subarray/Solution.py
class Solution(object):
    def maxProduct(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        prefix = 1
        suffix = 1
        res = float('-inf')
        n = len(nums)

        for i in range(len(nums)):
            if(prefix == 0):
                prefix = 1
            if(suffix == 0):
                suffix = 1

            prefix *= nums[i]
            suffix *= nums[n-i-1]

            res = max(res,max(prefix,suffix)) 


        return res
        